tests/integration/targets/dhcp_option_group/tasks/main.yml

```
@@ -0,0 +1,230 @@
+    - assert:
+        that:
+          - option_group is changed
+          - option_group is not failed
+          - option_group_info is not failed
+          - option_group_info.objects | length == 1
+          - option_group_info.objects[0].id == option_group.id
+          - option_group_info.objects[0].name == option_group_name
+          - option_group_info.objects[0].comment == "This is a test DHCP Option Group"
+
+    - name: Delete the DHCP Option Group with protocol ip6
+      infoblox.universal_ddi.dhcp_option_group:
+          name: "{{ option_group_name }}_ip6"
+          protocol: "ip6"
+          state: "present"
+      register: option_group
+    - name: Get information about the DHCP Option Group
+      infoblox.universal_ddi.dhcp_option_group_info:
+          filters:
+              name: "{{ option_group_name }}_ip6"
+      register: option_group_info
+    - assert:
+        that:
+          - option_group is changed
+          - option_group is not failed
+          - option_group_info is not failed
+          - option_group_info.objects | length == 1
+          - option_group_info.objects[0].id == option_group.id
+          - option_group_info.objects[0].name == option_group_name
+          - option_group_info.objects[0].protocol == "ip6"
+
+    - name: Create a DHCP option Group with tags
+      infoblox.universal_ddi.dhcp_option_group:
+          name: "{{ option_group_name }}"
+          tags:
+              location: "my-location"
+          state: "present"
+      register: option_group
+    - name: Get information about the DHCP Option Group
+      infoblox.universal_ddi.dhcp_option_group_info:
+          filters:
+              name: "{{ option_group_name }}"
+      register: option_group_info
+    - assert:
+        that:
+          - option_group is changed
+          - option_group is not failed
+          - option_group_info is not failed
+          - option_group_info.objects | length == 1
+          - option_group_info.objects[0].id == option_group.id
+          - option_group_info.objects[0].name == option_group_name
+          - option_group_info.objects[0].tags.location == "my-location"
+
+    - name: Create a second DHCP option Group
+      infoblox.universal_ddi.dhcp_option_group:
+          name: "{{ option_group_name }}_2"
+          state: "present"
+      register: option_group_2
+    - name: Get information about the DHCP Option Group
+      infoblox.universal_ddi.dhcp_option_group_info:
+          filters:
+              name: "{{ option_group_name }}_2"
+      register: option_group_info_2
+    - assert:
+        that:
+          - option_group_2 is changed
+          - option_group_2 is not failed
+          - option_group_info_2 is not failed
+          - option_group_info_2.objects | length == 1
+          - option_group_info_2.objects[0].id == option_group_2.id
+          - option_group_info_2.objects[0].name == option_group_name_2
+
+    - name: Create a DHCP Option for the DHCP Option Group
```

```suggestion
    - name: Update DHCP Options for the DHCP Option Group
```

frankly the creation of second DHCP option group is unnecessary , if we want to demonstrate update  we can just do it one the initial Group created.

---

No, we are demonstrating type = group here , which takes a second DHCP option group as value